### ISSUE TITLE
SVG icons instead of font icon?

### ISSUE THREAD

**Issue by olegstepura** [react-toolbox/react-toolbox#1174]
I suggest using packages like [react-icons](http://gorangajic.github.io/react-icons/go.html) + [react-svg-morph](http://gorangajic.github.io/react-svg-morph/) instead of font icon. Most applications don't need full package of icons, importing a component with only the required svg icon might be a better idea. This will result in less requests on page load (in case of webpack bundle), less bytes to download overall (hopefully).

Also this will eliminate the requirement for the user to add some extra tags in head (less requirements, be more modular)

What do you think?

**Comment by javivelasco** [comment]
Cool [react-svg-morph](http://gorangajic.github.io/react-svg-morph/)! I didn't now it. I don't like the svg font at all, but it's an implementation of material design icons. Maybe we can use https://google.github.io/material-design-icons/ some way.

**Comment by olegstepura** [comment]
Why don't you think `react-icons` can be used here? I suspect it is using svg files from `material-design-icons` under the hood.

**Comment by javivelasco** [comment]
Yeah, I missed the tab with material design icons. Being that the case I think it would be great. How would it be the integration for you? Would you map the name of material design icons some way? It can be backwards compatible that way

**Comment by lucasljj** [comment]
We can write a little module to map the material icons svg in a way that it can be backwards compatible with the web font

Preferably using a sprite https://github.com/jonathantneal/svg4everybody, it's safe to use that without a polyfill these days

**Comment by olegstepura** [comment]
I would just use the entire package. Without backward compatibility. `react-icons` offers more icons than Material Icons can offer. And that is cool. I was working on bootstrap project with FontAwesome icons, and you know, even FontAwesome does not contain icon for every use case.

**Comment by lucasljj** [comment]
Or with a js module that cache the used icons in variables, something like http://www.material-ui.com/#/components/svg-icon (see material icons example), but without needed to import each icon

**Comment by javivelasco** [comment]
Well, you can always allow to do something like:

```jsx
import { Icon } from 'react-toolbox';
import foo from 'react-ticons/lib/foo';

const UseIcon = () => <Icon value={foo} />;
```

This way you can by default allow to transparently use an icon with `<Icon value="code" />` and also you would pass an SVG getting the styles set by the component. Those styles could adapt the sizing to be used in a text context.

**Comment by olegstepura** [comment]
I tried using `react-icons` and seems like either I'm missing something, or it does not fit [Pixel perfection](https://github.com/react-toolbox/react-toolbox/blob/dev/ROADMAP.md#user-content-the-manifesto) manifesto. 

`react-icons`:
![2017-01-30 12_02_05-mahjong backend](https://cloud.githubusercontent.com/assets/534510/22418977/010fb9b4-e6e4-11e6-9893-ea240d08a6f1.png)

current solution with icon font:
![2017-01-30 12_02_30-mahjong backend](https://cloud.githubusercontent.com/assets/534510/22418984/03d69654-e6e4-11e6-8741-507911124e67.png)



Also [there is an issue](https://github.com/gorangajic/react-svg-morph/issues/22) that currently prevents usage of `react-icons-morph` with `react-icons`

**Comment by izakfilmalter** [comment]
The problem with `react-icons` is that they are not the exact icons.

![image](https://cloud.githubusercontent.com/assets/5874330/22435657/c30e60a6-e6d5-11e6-847e-4b21e762cc0f.png)

On the left I have the icon downloaded from https://material.io/icons/#ic_library_add
On the right we have the same icon from `react-icon`. As you can see for starters, `react-icons` is at 40dp which google doesn't produce:

![image](https://cloud.githubusercontent.com/assets/5874330/22435748/18d2130c-e6d6-11e6-99af-16c78ace7618.png)

Secondly, the path is all wrong. The Google version is significantly simpler.

**Comment by izakfilmalter** [comment]
I wouldn't mind spinning out a lib that contains all the svg icons from https://material.io/icons/ as react components.

**Comment by javivelasco** [comment]
I was about to say the same @izakfilmalter but Google should provide it though :/

**Comment by izakfilmalter** [comment]
They do, here: https://github.com/google/material-design-icons/. I was saying that I can go and create a middleware for it.

**Comment by javivelasco** [comment]
Oh cool. Makes sense to me, I way prefer SVG over a font and if react icons is not *exactly* the same as MD icons, I'd go that way

**Comment by JulianJorgensen** [comment]
Guys, is it possible to optionally remove Material Icons all together? FontAwesome 5 is in beta, and looking great! I've already integrated it using svg-sprite-loader. This seems to be the best way to integrate icons into react sites.

**Comment by olegstepura** [comment]
Some icons are used in the provided components. E.g. `DatePicker` has left right &  arrows which are Material Design icons.
![2017-08-10 09_58_44-react toolbox](https://user-images.githubusercontent.com/534510/29160439-836b6c9a-7db2-11e7-8371-ce57be3402a0.png)

**Comment by JulianJorgensen** [comment]
@olegstepura yes exactly. That's the icons that I'd like to somehow replace with my own internally hosted icons (FontAwesome in this case). I guess I could do it through the theme, right?

**Comment by olegstepura** [comment]
Not sure if you can replace this via CSS (theme) but you can do that via factoring out all components with a different implementation of `FontIcon` component. This isn't that easy. You can check sources on how to do that
